crates/compiler/gen_llvm/src/llvm/bitcode.rs

```diff
@@ -389,23 +388,23 @@ fn build_rc_wrapper<'a, 'ctx, 'env>(
             debug_info_init!(env, function_value);
 
             let mut it = function_value.get_param_iter();
-            let value_ptr = it.next().unwrap().into_pointer_value();
+            let generic_value_ptr = it.next().unwrap().into_pointer_value();
 
-            value_ptr.set_name(Symbol::ARG_1.as_str(&env.interns));
+            generic_value_ptr.set_name(Symbol::ARG_1.as_str(&env.interns));
 
-            let value_type = basic_type_from_layout(env, layout).ptr_type(AddressSpace::Generic);
+            let value_ptr_type =
+                basic_type_from_layout(env, layout).ptr_type(AddressSpace::Generic);
+            let value_ptr =
+                env.builder
+                    .build_pointer_cast(generic_value_ptr, value_ptr_type, "load_opaque");
 
+            // even though this looks like a `load_roc_value`, that gives segfaults in practice.
+            // I suspect it has something to do with the lifetime of the alloca that is created by
+            // `load_roc_value`
```

Comment:
That's interesting. Maybe load_roc_value isn't always safe? For example it doesn't look like it hides away the nullable variant in the lower bits of the alloca pointer when doing store_roc_value for recursive tag unions